### DISCUSSION TITLE
‎Issue with Search Bar and Search Results Web Parts in Modern SharePoint Site

### DISCUSSION THREAD

**Discussion by Tishaay** [microsoft-search/pnp-modern-search#3963] [General]
We are currently in the process of developing a Modern SharePoint site and have encountered a significant issue related to the search functionality. Specifically, whenever we configure the search bar web part and attempt to connect it to the search results web part, the entire page loads but then becomes unresponsive. This unresponsiveness occurs consistently, preventing us from making any further changes or interacting with the page.

In the past, configuring the search bar and search results web parts was a straightforward process, and we did not encounter any issues. However, we’ve also observed that the search bar and search results web parts that were previously configured and functioning correctly are now non-operational. These web parts, which worked seamlessly before, are no longer returning any results or responding as expected.

At present, we are unable to connect the search box web part to the search results web part, as doing so causes the entire site page to hang and become unresponsive.

Given the importance of the search functionality to our site, this issue is a significant concern. We would greatly appreciate any insights or guidance on how to resolve these problems and restore the functionality of both the newly configured and previously working search web parts.

**Comment by kasperbolarsen** [comment]
We can not give you any guidance at all based on your description as it contains zero specifics. Please update with the technical details you would expect if I was asking you for guidance :-)

**Reply by Tishaay** [reply]
Hi, we can't connect the search box webpart to the search results webpart because whenever we connect it the site page hangs up and becomes unresponsive.

Here is a screenshot of our issue.
![image](https://github.com/user-attachments/assets/82c9c1ea-c8d6-4bcb-8744-311869e734eb)

**Reply by kasperbolarsen** [reply]
Which version of PnP Modern Search?
Which browser?
Have you tried so set the search query up in the results web part and then connect from the Results web part to the search box?

**Reply by Tishaay** [reply]
I'm uncertain about the exact version being used. I have utilized the web part available in Modern SharePoint and configured it accordingly using the Microsoft Edge browser.
![image](https://github.com/user-attachments/assets/8402b4b2-a20f-41d1-97f0-5928a4e17072)

When the search box web part is configured and the search results property is modified, the page becomes unresponsive.

**Reply by kasperbolarsen** [reply]
Open the configuration of the Results web part  and go to page 4, the version number should be visible there

**Reply by Tishaay** [reply]
There are only 3 pages in the Search results configurations.
![image](https://github.com/user-attachments/assets/801d96dd-fcc0-48ec-bb83-defb4c6a0c46)

**Reply by kasperbolarsen** [reply]
That would make it a version 3.* which was deprecated some years ago. 
I would recommend having one of your SharePoint admins uninstall that version in the App catalog,  and install the current version,  4.13 1 from 
https://github.com/microsoft-search/pnp-modern-search/releases/tag/4.13.1

Please be advised that a v3 cannot be upgraded to v4. , you will have to set up the web parts again

**Reply by Tishaay** [reply]
Hi @kasperbolarsen, where should we install this? Is this done in a tenant level in SharePoint?

**Reply by kasperbolarsen** [reply]
Hi, normally the package is installed in the Tenant level App catalog, but it is also possible to install it in a Site Collection App Catalog, once this has been created.
see https://microsoft-search.github.io/pnp-modern-search/

**Reply by Tishaay** [reply]
Could the issue of the search box becoming unresponsive and hangs up when connected to the search results be due to the installed package not being updated to version 4?
